src/index.ts

```diff
@@ -186,6 +186,7 @@ export function titleCase<
 >(str?: T, opts?: UserCaseOptions) {
   return (Array.isArray(str) ? str : splitByCase(str as string))
     .filter(Boolean)
+    .map((p) => p.trim())
```

Comment:
While this fixed the issue, adds overhead to the operation. I am wondering if we could fix it properly on `splitByCase` internals

Reply:
Moved `trim` to `splitByCase` function.